src/components/Menu/index.tsx

```diff
@@ -0,0 +1,186 @@
+                <div style={{ width: '50%', marginLeft: '30px' }}>
+                  <Text noSelect fontFamily="Nunito Bold" fontSize="32px">
+                    {t('menu:time')}
+                  </Text>
+                  <TimePicker programTime={time} />
+                </div>
+              </div>
+              <Text noSelect fontFamily="Nunito Bold" fontSize="32px">
+                {t('menu:length')}
+              </Text>
+              <DurationPickers programDuration={timeLength} />
+            </Form>
+          </FormContainer>
+        </BottomContainer>
+        <ButtonContainer>
+          <Button
+            margin="auto 0 0 0"
+            text={t('menu:cancel')}
+            icon={<CgClose />}
+            onClick={() => {
+              setisClosing(true);
+            }}
+          />
+          <Button
+            margin="auto 0 0 27px"
+            text={t('menu:save')}
+            icon={<AiOutlineSave />}
+            onClick={() => {
+              let desc;
+              if (document.getElementsByTagName('textarea')[0] !== undefined) {
+                desc = document.getElementsByTagName('textarea')[0]?.innerHTML;
+              }
+              let dur;
+              time = document.getElementsByTagName('input')[6].value;
+              // eslint-disable-next-line no-restricted-globals
+              if (isNaN(dur)) {
+                dur =
+                  time.split('hour')[0] * 3600 +
+                  time.split('hour')[1].split('minutes')[0] * 60;
+              } else {
+                dur =
+                  time.split('hours')[0] * 3600 +
+                  time.split('hours')[1].split('minutes')[0] * 60;
+              }
+              const hour = new Date();
+              const instant = document.getElementsByTagName('input')[5].value;
+              hour.setHours(
+                Number(instant.split(':')[0]),
+                Number(instant.split(':')[1]),
+                Number(instant.split(':')[2]),
+              );
+              const rate = document.getElementsByTagName('input')[3].value;
+              const ratings = {
+                RL: ProgramRating.RL,
+                R10: ProgramRating.R10,
+                R12: ProgramRating.R12,
+                R14: ProgramRating.R14,
+                R16: ProgramRating.R16,
+                R18: ProgramRating.R18,
+              };
+              programs[aux].title =
+                document.getElementsByTagName('input')[2].value;
+              programs[aux].description = desc;
+              programs[aux].rating = ratings[rate];
+              programs[aux].startDate = new Date(
+                document.getElementsByTagName('input')[4].value,
+              );
+              programs[aux].startHour = new Date(hour);
+              programs[aux].duration = Number(dur);
+              setProgram(programs);
+              // console.log(aux);
+              // console.log(program);
+              setisClosing(true);
+            }}
```

Comment:
`<Menu />` should receive a `onSaveProgram` callback prop, which is triggered here instead.

```suggestion
            onClick={() => onSaveProgram(newProgram)}
```